src/qibolab/couplers.py

```diff
@@ -0,0 +1,58 @@
+from dataclasses import dataclass, field
+from typing import List, Optional, Union
+
+from qibolab.channels import Channel
+from qibolab.native import TwoQubitNatives
+from qibolab.qubits import Qubit
+
+CouplerId = Union[str, int]
+"""Type for Coupler names."""
+
+
+@dataclass
+class Coupler:
+    """Representation of a physical coupler.
+
+    Coupler objects are instantiated by :class:`qibolab.platforms.platform.Platform`
+    but they are passed to instrument designs in order to play pulses.
+
+    Args:
+        name (int, str): Qubit number or name.
+
+        flux (:class:`qibolab.platforms.utils.Channel`): Channel used to
+            send flux pulses to the qubit.
+        Other characterization parameters for the coupler, loaded from the runcard.
+    """
+
+    name: CouplerId
+
+    frequency: int = 0
+    sweetspot: float = 0
+
+    flux: Optional[Channel] = None
+    qubits: Optional[List["Qubit"]] = field(default_factory=dict)
```

Comment:
The `Coupler`s are already "hanging" from the `Qubit`s they are connected to. Do the `Qubit`s also need to hang from the `Coupler`s? This is the cyclic thing I was talking about. It won't cause any error, we just avoid it as much as possible from simplicity.

Reply:
That's true I still not sure what would be the best for doing 2q gate routines with couplers